website/notifications/model.py

```diff
@@ -4,18 +4,35 @@
 from modularodm.exceptions import ValidationValueError
 
 from website.project.model import Node
-from website.notifications.constants import NOTIFICATION_TYPES
+from website.notifications.constants import NOTIFICATION_TYPES, NODE_SUBSCRIPTIONS_AVAILABLE, USER_SUBSCRIPTIONS_AVAILABLE
+import re
 
 
 def validate_subscription_type(value):
     if value not in NOTIFICATION_TYPES:
         raise ValidationValueError
 
 
+def validate_event_type(value):
+    """
+
+    :param value:
+
+    Checks if the value passed in is in the NODE_SUBSCRIPTIONS_AVAILABLE or USER_SUBSCRIPTIONS_AVAILABLE
+    if it is an event name.  Since the only things from NODE_SUBSCRIPTIONS_AVAILABLE that contains "file_updated"
+    is preceded by a user ID, a regex checker was added for that.
+    """
+    prefix = re.compile('[[a-zA-Z0-9]*_]?')
```

Comment:
I don't believe it would.  The regex I entered can only check for a user ID
and an underscore (at least I think).  For the example you gave, since it
has more than one underscore (the regex is limited to just 0 or 1 to
check), it will not match.  Then it will search it normally in the
NODE/USER_SUBSCRIPTIONS_AVAILABLE dictionary and correctly match it.
Through all of the tests, I found that the only values in those two
dictionaries that can be preceded by a user ID is the "file_updated", so
that is why it only checks and fixes that one.

On Wed, Aug 3, 2016 at 4:52 PM, Brian J. Geiger <notifications@github.com>
wrote:

> In website/notifications/model.py
> <https://github.com/CenterForOpenScience/osf.io/pull/6026#discussion_r73418483>
> :
>
> >
> >
> >  def validate_subscription_type(value):
> >      if value not in NOTIFICATION_TYPES:
> >          raise ValidationValueError
> >
> >
> > +def validate_event_type(value):
> > +    """
> > +
> > +    :param value:
> > +
> > +    Checks if the value passed in is in the NODE_SUBSCRIPTIONS_AVAILABLE or USER_SUBSCRIPTIONS_AVAILABLE
> > +    if it is an event name.  Since the only things from NODE_SUBSCRIPTIONS_AVAILABLE that contains "file_updated"
> > +    is preceded by a user ID, a regex checker was added for that.
> > +    """
> > +    prefix = re.compile('[[a-zA-Z0-9]*_]?')
>
> So if the value were global_comment_replies, would that match?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CenterForOpenScience/osf.io/pull/6026/files/9f6320d17d9ba489ffddce6590144d5e9c3091e6#r73418483>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AP4VDZ8YlZrZWTiUrk0PMynjOKqF7dVEks5qcP-OgaJpZM4JUReH>
> .
>



-- 
Daniel Brown
University of Virginia
Computer Science
Class of 2018